recipes/libelfin/all/patches/946dde57e5caef8297b9339f3a7971401d540840.patch

```diff
@@ -0,0 +1,165 @@
+diff --git a/dwarf/abbrev.cc b/dwarf/abbrev.cc
+index c780b8e..f77dc0c 100644
+--- a/dwarf/abbrev.cc
++++ b/dwarf/abbrev.cc
+@@ -133,6 +133,10 @@ resolve_type(DW_AT name, DW_FORM form)
+                 case DW_AT::ranges:
+                         return value::type::rangelist;
+ 
++                case DW_AT::lo_user...DW_AT::hi_user:
++                        //HACK: ignore vendor extensions
++                        return value::type::invalid;
```

Comment:
Where does this patch come from?

Reply:
I think there is a better way to do this...
coz recommends using libelfin v0.2, but the only libelfin version I found is working with coz is the latest. Unfortunately, it wasn't properly released. There are a few commits on top of libelfin v0.3. So the patch does the fix for some files to that commit. I can change the recipe to use git.clone to get the latest, but what would I use for the version? Using the `master` from libelfin will have an issue with build reproducibility, so we can use specific hash. But how do you choose the version tag for it? `0.3-1`?

Reply:
Perhaps just take those commit(s) and apply them and reference them as a comment on top of the patch?

we have "virtual versioning" which is `cci.yyyymmdd` but I dont think we need that here, try the suggestion above?

Reply:
Reduced patch to the critical fixes and added comment:
~~~
> head patches/9d0db16d0a0b3c4f8aaa60a3e4dab295df34b6b2.patch
#
# Updates v0.3 to the commit 9d0db16d0a0b3c4f8aaa60a3e4dab295df34b6b2:
#   9d0db16 fix data initialization of elf segment
#   74f1116 ignore dwarf extensions of gcc in the line table
#   ac45a09 Added #include <errno.h> to files that use errno.
#   4a678c7 Don't mandate that compilation units need a DW_AT_comp_dir for line table parsing
#   85812f6 Fix typos
#
~~~